aws_lambda_builders/workflows/nodejs_npm/npm.py

```diff
@@ -0,0 +1,90 @@
+"""
+Wrapper around calling npm through a subprocess.
+"""
+
+import logging
+
+LOG = logging.getLogger(__name__)
+
+
+class NpmExecutionError(Exception):
+
+    """
+    Exception raised in case NPM execution fails.
+    It will pass on the standard error output from the NPM console.
+    """
+
+    MESSAGE = "NPM Failed: {message}"
+
+    def __init__(self, **kwargs):
+        Exception.__init__(self, self.MESSAGE.format(**kwargs))
+
+
+class SubprocessNpm(object):
+
+    """
+    Wrapper around the NPM command line utility, making it
+    easy to consume execution results.
+    """
+
+    def __init__(self, osutils, npm_exe=None):
+        """
+        :type osutils: aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils
+        :param osutils: An instance of OS Utilities for file manipulation
+
+        :type npm_exe: str
+        :param npm_exe: Path to the NPM binary. If not set,
+            the default executable path npm will be used
+        """
+        self.osutils = osutils
+
+        if npm_exe is None:
+            if osutils.is_windows():
+                npm_exe = 'npm.cmd'
```

Comment:
Does npm have a `npm.exe` at all?

Reply:
no it doesn't unfortunately. I verified on a windows box